GDScript: Fix for UNUSED_PRIVATE_CLASS_VARIABLE

ZuBsPaCe opened the pull request:
Fixes following issues in master:

Fixes #47888
Fixes #42318

Usage count of member variables was never increased. This lead to UNUSED_PRIVATE_CLASS_VARIABLE warnings for private variables like "_my_private_var" even if the variable was used.

Test script, which will only report "_private_unused" with this fix:

```gdscript
var _private_used: int = 1
var _private_unused: int = 1

var public_used: int = 1
var public_unused: int = 1

func test() -> void:
	public_used = _private_used
```

ZuBsPaCe commented:
Was a bit unsure, if I should use a switch statement on ClassNode::Member instead. Not sure, if handling all cases is necessary, but I don't think so. Decided to use the simple if statement just for ClassNode::Member::VARIABLE.

vnen commented on modules/gdscript/gdscript_parser.cpp:
```diff
@@ -2042,6 +2042,14 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_identifier(ExpressionNode
+			identifier->source = IdentifierNode::MEMBER_VARIABLE;
+			identifier->variable_source = declaration.variable;
+			declaration.variable->usages++;
+		}
```
This won't really work in all cases because member variables can be used before they are declared. This check needs to be done in the analyzer.

ZuBsPaCe replied:
You're right, thanks! Did not think about that. Moved the usage increment into the analyzer. Here's a small test case:
```gdscript
var pub

var _priv_forward_used
var _priv_forward_unused

func test1() -> void:
	pub = _priv_forward_used
	pub = _priv_backward_used
	
var _priv_backward_used
var _priv_backward_unused
```
Result:
![image](https://user-images.githubusercontent.com/4233942/131865563-83a7cd47-5230-4eef-910a-092c75cebe29.png)

I don't like, that some usages are counted in the parser and this new one in the analyzer...

vnen replied:
> I don't like, that usages are counted in the parser and this new one in the analyzer, but I don't think that this could be changed easily..

TBH I don't like it either, but for now I guess it's not worth changing.

vnen reviewed (approved):
Looks good to me.

vnen commented:
@ZuBsPaCe I just noticed that the commit is being attributed to another user. You may want to change the email address in your git settings to one that is tied to your user in GitHub and then run `git commit --amend --no-edit --reset-author` to change the author in the commit and force push to update the PR.

ZuBsPaCe committed:
GDScript: Count usages of member variables.

Otherwise private member variables will always lead to UNUSED_PRIVATE_CLASS_VARIABLE.

ZuBsPaCe commented:
@vnen Thanks for pointing this out to me! Completely missed that one. 

Now everything should be correct.

mhilbrunner commented:
Built and tested, works and indeed fixed the issue which I could reproduce without this.

Thanks for the fix!
